src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -141,7 +142,7 @@ private void GenerateProlog()
             int totalLocalSize = 0;
             foreach(LocalVariableDefinition local in _locals)
             {
-                int localSize = local.Type.GetElementSize().AsInt;
+                int localSize = local.Type.GetElementSize().AsInt.AlignUp(_pointerSize);
```

Comment:
Alignment should depend on the type to avoid being overly inefficient. In other words, if you have two ```byte``` locals, they should be at offsets 0 and 1, if you have two ```char``` locals, they should be at offsets 0 and 2. We likely need to align up based on the variable size and current stack position. For example, if there is a ```byte```, followed by a ```char```, they should be at 0 and 2 (with 1-byte padding at offset 1), if there is a ```byte``` followed by an ``int```, they should be at 0 and 4.

I think ```MetadataFieldLayoutAlgorithm.ComputeSequentialFieldLayout``` has similar logic to what we need here -- it aligns up the position based on the alignment requirements of the current type.

Reply:
Ok, I think I've matched the padding/alignment logic from `MetadataFieldLayoutAlgorithm.ComputeSequentialFieldLayout`, I ignored quite a bit of extra stuff going on in there that looked specific to type layout let me know if I ignored too much.